changelog.d/15079.bugfix

```diff
@@ -0,0 +1 @@
+Fix a bug introduced in Synapse v1.74.0 where searching with colons when using ICU for search term tokenisation would fail with an error.
```

Comment:
(Adding a thread at a random spot...)

Did you [consider using the `websearch_to_tsquery`function](https://github.com/matrix-org/synapse/issues/14815#issuecomment-1378822009) at all? I'm not sure it would really solve this problem (or how it would interact with ICU), but just curious mostly!

Reply:
I did, but we intentionally don't use that function because we want to support prefix searching (i.e. if you search `bo` you find `bob`). `websearch_to_tsquery` gives you an effortless traditional search-engine-style syntax which is very nice but it doesn't do that.